### PR TITLE
move the generation of the python stubs to the api repo and modify ac…

### PR DESCRIPTION
**Pull request by mariostieriansys**
The python stubs are now generated as part of the api repo build process via a modified sdist command that executes the generation.
In stub_api.py the imports created in the Python files have been modified accordingly to the path of installation of the api repository.

Finally, the test assets generation is now included as part of the wheel, so it will have to be consumed from the api repo installation location rather than the tests location in the pyensight repo

### CODE REVIEW

**Review by AlejandroFernandezLuces — APPROVED** [review]
Code LGTM, just some minor comments. Thanks a lot for the work!

I was wondering how the documentation generation will work. How the documentation that is generated in this script is taken to the main repository?

**Comment by mariostieriansys** [comment]
@AlejandroFernandezLuces the process is this:

1. In EnSight we have some Python native APIs, and each method/class has got their own documentation/docstrings
2. These are dumped into the XML file, which lists all these methods and classes
4. The stub generator creates the modules parsing the xml. Within the stub generator call, we have a "replacement" class which parses the content in docs/replacement and replaces the docs we get from the xml in the newly generated modules.


Both the XML "docs" and the "replacement" docs will become docstrings in the auto-generated modules. These docstrings will be then parsed during the PyEnSight Sphinx Doc build. 

So at this level what we are generating is just docstrings, with the replacement just replacing the docstrings we would get from the XML file. The actual docs generation happens in PyEnSight, which parses the docstrings of the autogenerated files.

**Comment by AlejandroFernandezLuces** [comment]
> @AlejandroFernandezLuces the process is this:
> 
> 1. In EnSight we have some Python native APIs, and each method/class has got their own documentation/docstrings
> 2. These are dumped into the XML file, which lists all these methods and classes
> 3. The stub generator creates the modules parsing the xml. Within the stub generator call, we have a "replacement" class which parses the content in docs/replacement and replaces the docs we get from the xml in the newly generated modules.
> 
> Both the XML "docs" and the "replacement" docs will become docstrings in the auto-generated modules. These docstrings will be then parsed during the PyEnSight Sphinx Doc build.
> 
> So at this level what we are generating is just docstrings, with the replacement just replacing the docstrings we would get from the XML file. The actual docs generation happens in PyEnSight, which parses the docstrings of the autogenerated files.

I see, it's just for the docstrings, got it!

File: setup.py [hunk 1]
```diff
@@ -23,6 +29,44 @@
 
 description = f"Autogenerated python gRPC interface package for {package_name}, built on {datetime.now().strftime('%H:%M:%S on %d %B %Y')}"
 
+class CodegenCommand(distutils.cmd.Command):
+    """A custom command to runthe generation of the Python stubs."""
```
[the remark this answers is not in the document]
Comment (mariostieriansys) [reply setup.py]: fixed it in a new commit

File: codegen/generate.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import stub_api
+
+
+if __name__ == "__main__":
+    stub_api.generate()
```
[the remark this answers is not in the document]
Comment (mariostieriansys) [reply codegen/generate.py]: good suggestion, fixed in the new commit

**Comment by mariostieriansys** [comment]
@AlejandroFernandezLuces just a few things to note when you use the new changes in PyEnSight:

1. All the imports of the autogenerated objects will need to be modified to take into account of the new path of the stubs, that is now ansys.api.pyensight and no more just ansys.pyensight.
2. The stubs also generate a txt file, that is called "ensight_api_test_assests.txt". This is part of PyEnSight under "tests/unit_tests" and is consumed by "tests/unit_tests/test_ensight_api.py".
Since this is now generated as part of the API repo, the test in "tests/unit_tests/test_ensight_api.py" will need to load the file from the installation location, ansys/api/pyensight/ensight_api_test_assests.txt and no more from tests/unit_tests/ensight_api_test_assests.txt (I am talking about line 13 of test_ensight_api.py). This is important otherwise the tests will fail

**Pull request merged**